ge/org.osate.ge/src/org/osate/ge/internal/ui/properties/ChangeSubcomponentTypePropertySection.java

```diff
@@ -140,11 +141,7 @@ public void refresh() {
 	}
 
 	private static boolean isCompatibleSupcomponentType(final NamedElement ne, final EClass subcomponentType) {
-		Subcomponent sc = (Subcomponent) ne;
-		if (sc.getRefined() != null) {
-			sc = sc.getRefined();
-		}
-
+		final Subcomponent sc = (Subcomponent) AadlHelper.getRootRefinedElement(ne);
```

Comment:
I'll try to ignore problems like the fact `getTypeOptionsInformation` does something different from its name IMHO, which are outside the scope of this issue, I don't believe this fix is what we want.

I'm having trouble understanding why the original code gets the refined element at the start. It would seem that the  if statement should be removed.  The compatible subcomponent should be based on the containing classifier of the subcomponent(even if it is refined) and if a AbstractSubcomponent is being refined, then it should be changeable to anything. 

In short, please avoid getting the refined subcomponent at the start and test.

Reply:
@RyanMcilnay sorry I added the last comment after you made changes. I should of taken my own advise and used the review feature.